src/layout/layout.rs

```diff
@@ -837,29 +849,30 @@ fn configure_fill_constraints(
     constraints: &[Constraint],
     flex: Flex,
 ) -> Result<(), AddConstraintError> {
-    for ((&left_constraint, &left_element), (&right_constraint, &right_element)) in constraints
-        .iter()
-        .zip(segments.iter())
-        .filter(|(c, _)| c.is_fill() || (!flex.is_legacy() && c.is_min()))
-        .tuple_combinations()
-    {
-        let left_scaling_factor = match left_constraint {
-            Constraint::Fill(scale) => f64::from(scale).max(1e-6),
-            Constraint::Min(_) => 1.0,
-            _ => unreachable!(),
-        };
-        let right_scaling_factor = match right_constraint {
-            Constraint::Fill(scale) => f64::from(scale).max(1e-6),
-            Constraint::Min(_) => 1.0,
-            _ => unreachable!(),
-        };
-        solver.add_constraint(
-            (right_scaling_factor * left_element.size())
-                | EQ(GROW)
-                | (left_scaling_factor * right_element.size()),
-        )?;
-    }
-    Ok(())
+    tuple_combinations(
```

Comment:
Often I prefer the functional expressions, but I think the for loop here is much better as it the functional approach seems more difficult to reason about.